File: leetcode/longest-substring-without-repeating-characters.cpp
#include "string"
#include "iostream"
using namespace std;

class Solution
{
public:
  int lengthOfLongestSubstring(string s)
  {
    int len = 0, maxLen = 0;
    auto check = [&len, &maxLen, &s](char chr, int end) -> void
    {
      for (int i = end - len; i < end; i++)
      {
        if (s[i] != chr)
          continue;
        if (len > maxLen)
          maxLen = len;
        len = end - i - 1;
      }
    };
    for (int i = 0; i < s.length(); i++)
    {
      check(s[i], i);
      len++;
    }
    if (len > maxLen)
      maxLen = len;
    return maxLen;
  }
};

File: leetcode/longest-substring-without-repeating-characters.js
/**
 * @param {string} s
 * @return {number}
 */
 var lengthOfLongestSubstring = function (s) {
  let len = 0;
  let maxLen = 0;
  for (let i = 0; i < s.length; i++) {
      check(s[i], i);
      len++;
  }
  if (len > maxLen) maxLen = len;
  return maxLen;
  function check(chr, end) {
      for (let i = end - len; i < end; i++) {
          if (s[i] != chr) continue;
          if (len > maxLen) maxLen = len;
          len = end - i - 1;
          break;
      }
  }
};